### PR TITLE
fix: nil pointer exception in diff calculation

### LINKED ISSUE #434: [Regression] invalid memory address or nil pointer dereference
After updating from v0.7.18 to v0.7.19 I have this error:

```
An unexpected error occurred
runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0xa5b740, 0xf88470, 0x0)
        /opt/hostedtoolcache/go/1.15.8/x64/src/runtime/debug/stack.go:24 +0x9f
main.handleUnexpectedErr(0xc00016a790, 0xc0001da4e0, 0xa84580, 0xf418c0)
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:179 +0x1c5
main.main.func1(0xc0000a5f40, 0xc00016a790, 0xc0000a5ef8, 0xc000114300)
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:50 +0xba
panic(0xa84580, 0xf418c0)
        /opt/hostedtoolcache/go/1.15.8/x64/src/runtime/panic.go:969 +0x1b9
github.com/infracost/infracost/internal/schema.diffResourcesByKey(0x0, 0x0, 0xc0000a5630, 0xc0000a5600, 0x0, 0xc000621180)
        /home/runner/work/infracost/infracost/internal/schema/diff.go:70 +0x174
github.com/infracost/infracost/internal/schema.calculateDiff(0xc0001b4800, 0x28, 0x40, 0xc0001b4400, 0x28, 0x40, 0x0, 0x1, 0x10000c00016a6e0)
        /home/runner/work/infracost/infracost/internal/schema/diff.go:35 +0x205
github.com/infracost/infracost/internal/schema.(*Project).CalculateDiff(...)
        /home/runner/work/infracost/infracost/internal/schema/project.go:29
main.defaultMain(0xc00016a790, 0x0, 0x0)
        /home/runner/work/infracost/infracost/cmd/infracost/default.go:300 +0x94b
main.defaultCmd.func1(0xc000229000, 0xc000264900, 0x0)
        /home/runner/work/infracost/infracost/cmd/infracost/default.go:130 +0x105
github.com/urfave/cli/v2.(*App).RunContext(0xc0001da4e0, 0xbfd140, 0xc000138010, 0xc0001161b0, 0x1, 0x1, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:322 +0x70d
github.com/urfave/cli/v2.(*App).Run(...)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:122 +0x577

Environment:
Infracost v0.7.19
```

Please give me some advice to create a useful ticket.

### PR DESCRIPTION
**Pull request by sinabakh**
Fix #434

### CODE REVIEW

**Review by aliscott — APPROVED** [review]

File: internal/schema/diff.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func calculateDiff(past []*Resource, current []*Resource) []*Resource {
 func diffResourcesByKey(resourceKey string, pastResMap, currentResMap map[string]*Resource) (bool, *Resource) {
 	past, pastOk := pastResMap[resourceKey]
 	current, currentOk := currentResMap[resourceKey]
+	if current == nil && past == nil {
+		return false, nil
```
Comment (aliscott) [inline internal/schema/diff.go:61]: Should we add logging here to output the resourceKey so we can debug if this happens again?

Comment (sinabakh) [reply internal/schema/diff.go:61]: That would be great. However, we can't debug it just be this log and if the user provides its config then we can reproduce it. But I think it's better to add logs here and in the same location in the `diffCostComponentsByKey` function.

Comment (sinabakh) [reply internal/schema/diff.go:63]: @aliscott Added some debug info. Do we need more data in the log?

File: internal/schema/diff.go [hunk 2]
```diff
@@ -147,6 +152,7 @@ func diffCostComponentsByKey(key string, pastCCMap, currentCCMap map[string]*Cos
 	past, pastOk := pastCCMap[key]
 	current, currentOk := currentCCMap[key]
 	if current == nil && past == nil {
+		log.Debugf("diffResourcesByKey nil current and past with key %s", key)
```
Comment (aliscott) [inline internal/schema/diff.go:155]:
```suggestion
		log.Debugf("diffCostComponentsByKey nil current and past with key %s", key)
```

**Pull request merged**